### ISSUE TITLE
Hot Button not working

### ISSUE THREAD

**Issue by nikshriv** [nikshriv/hass_navien_water_heater#13]
> > The integration is done. Please download and give it a try. Let me know if you have any issues.
> 
> Hi Nik,
> 
> I was able to install it and it seems to be working. The last question during setup asked how often I'd like to poll the hub. So is this integration communicating with the hub directly, and not going through the cloud?
> 
> Edit: I just tried the hot button provided by it, but it doesn't seem to be doing anything.

Nik, Thanks for the work on this! 

Same issue here, everything except the Hot Button seems to be working. Hot Button status changes when activated in the Navien App, however it does not activate when turned on from HomeAssistant.

_Originally posted by @Pelphry17 in https://github.com/nikshriv/hass_navien_water_heater/issues/10#issuecomment-1455118905_

**Comment by nikshriv** [comment]
I'm not sure why it's not working. I don't use the hot button functionality myself, so I haven't tested it. I will test it on my own water heater to see what's wrong.

**Comment by megaz555** [comment]
Same issue here. The integration is working, but hot button is not activated. If you tap the button in HA, it turns on, then immediately turns off. The Navien app does not see this trigger and the hot button link never turns red.

Here is what it is supposed to look like:
![6420E736-0F73-4DD0-8AC9-824451720DDC](https://user-images.githubusercontent.com/24313201/222976536-cc8d2f1e-56bd-49a8-94a8-b3b6372774dc.jpeg)

Also, the previous integration that used to work would keep the button on in HA, and only turn off once the water recirculation was done. I don’t know if it is possible to do this with the new integration, but that would be an extremely helpful functionality to keep.

**Comment by nikshriv** [comment]
Fixed in the latest update. Please redownload and try again.

**Comment by Pelphry17** [comment]
> Fixed in the latest update. Please redownload and try again.

Confirmed. Hot Button function is working now. Thank You!

**Comment by megaz555** [comment]
Just tested it on my end. It is working! Not only that, but the functionality is exactly as it used to be - when the hot button is triggered in HA through a button, it stays lit up while recirculation is running. As soon as it is done, the button turns off.

@nikshriv, you sir are a wizard! Thank you for working hard on this integration to make it work. Thanks to people like you, the world is a better place for everyone.

**Comment by kchiem** [comment]
Do I just re-download in HACS to update?  Or do I need to delete and re-add the integration as well?

Also, can you answer this question?

> The last question during setup asked how often I'd like to poll the hub. So is this integration communicating with the hub directly, and not going through the cloud?

**Comment by megaz555** [comment]
There is an easy way to get HACS to recognize the update - just restart Home Assistant. Once HACS loads up, it will see the latest change in the repository and allow you to perform an upgrade.

As for the polling interval, that's how often you want the plugin to poll NaviLink. I would set it to something like 15 seconds, which I believe is the default. No, there is no way for the integration to communicate directly with the hub. NaviLink connects to Navien cloud service, so this integration connects to that cloud service to get the data. If your Internet connection goes down or Navien's cloud service is unreachable, the integration will stop working.

If you want to have a working solution even without the Internet, the only way is to add a hot button trigger directly on your unit. I went ahead and added a Z-Wave relay to mine, so that I can trigger recirculation with or without Internet connection (which I described how to do before in one of my posts).

Hope this makes sense.

**Comment by kchiem** [comment]
> If you want to have a working solution even without the Internet, the only way is to add a hot button trigger directly on your unit. I went ahead and added a Z-Wave relay to mine, so that I can trigger recirculation with or without Internet connection (which I described how to do before in one of my posts).

Yes, I saw that.  I also replied in that thread that I did the same, but with a Shelly 1, and asked you some follow up questions regarding actual functionality.  But you didn't seem to have noticed that before that issue was closed.  Anyways, I've resolved those performance issues and posted tips on the HA forum's Navilink thread, where I saw this integration mentioned, and where they're likely to be more visible long term.

I mean to ask nik about the polling because the wording on that question suggests that it might be getting updates locally, which would be great if that were the case.

**Comment by nikshriv** [comment]
megaz555 is right. The integration polls an amazon iot MQTT server which is what the Navilink device connects to and sends updates. The amazon iot server is an MQTT broker. So, this integration is dependent on an internet connection.

**Comment by kchiem** [comment]
Can confirm Hot Button works now as well.
